### PR TITLE
Adjust block creation to account for freeze times

### LINKED ISSUE #8176: Figure out why the record file hash chain is broken

### PR DESCRIPTION
**Pull request by mhess-swl**
This PR adds a condition that will trigger creating a new block of records, specifically by taking into account the last and current freeze time of the dual state. This is for when the node enters a state where the last and current freeze times are equal, as can be the case around node shutdown. 

Fixes #8176

### CODE REVIEW

**Review by tinker-michaelj — CHANGES_REQUESTED** [review]
We need to only use the new condition in `willCreateNewBlock()` when the `freezeTime` is non-null.

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/logic/BlockManager.java [hunk 1]
```diff
@@ -189,7 +194,15 @@ private void computeProvisionalBlockMeta(final Instant now) {
     private boolean willCreateNewBlock(@NonNull final Instant timestamp) {
         final var curNetworkCtx = networkCtx.get();
         final var firstBlockTime = curNetworkCtx.firstConsTimeOfCurrentBlock();
-        return firstBlockTime == null || !inSamePeriod(firstBlockTime, timestamp);
+        final var dualState = dualStateAccessor.getDualState();
+        final var lastFrozenAndCurrentFreezeTimeEqual =
```
[the remark this answers is not in the document]
Comment (mhess-swl) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/logic/BlockManager.java:197]: Fixed

**Review by tinker-michaelj — APPROVED** [review]
LGTM 🥂

**Review by jsync-swirlds — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Review by jsync-swirlds — APPROVED** [review]

**Pull request merged**